cli/main.rs

```diff
@@ -0,0 +1,117 @@
+use async_std::{
+    io::{self, Read, Write},
+    task,
+};
+use core::{fmt, str::FromStr};
+use exitfailure::ExitFailure;
+use failure::{format_err, Error};
+use multibase::Base;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+struct Opts {
+    /// The mode
+    #[structopt(subcommand)]
+    mode: Mode,
+}
+
+#[derive(StructOpt, Debug)]
+enum Mode {
+    #[structopt(name = "encode")]
+    Encode {
+        /// The base to use for encoding.
+        #[structopt(short = "b", long = "base", default_value = "base58-btc")]
+        base: StrBase,
+    },
+    #[structopt(name = "decode")]
+    Decode,
+}
+
+fn main() -> Result<(), ExitFailure> {
+    env_logger::init();
+    task::block_on(async {
+        let opts = Opts::from_args();
+        match opts.mode {
+            Mode::Encode { base } => encode(base).await,
+            Mode::Decode => decode().await,
+        }
+    })
+}
+
+#[derive(Debug)]
+struct StrBase(Base);
+
+impl fmt::Display for StrBase {
```

Comment:
maybe use [strum](https://github.com/Peternator7/strum) to avoid all the repetition below?